router/usecase/pools/routable_concentrated_pool.go

```diff
@@ -192,11 +193,136 @@ func (r *routableConcentratedPoolImpl) CalculateTokenOutByTokenIn(ctx context.Co
 	return sdk.Coin{Denom: tokenOutDenom, Amount: amountOutTotal.TruncateInt()}, nil
 }
 
+// CalculateTokenInByTokenOut implements domain.RoutablePool.
+// It calculates the amount of token out given the amount of token in for a concentrated liquidity pool.
+// Fails if:
+// - the underlying chain pool set on the routable pool is not of concentrated type
+// - fails to retrieve the tick model for the pool
+// - the current tick is not within the specified current bucket range
+// - tick model has no liquidity flag set
+// - the current sqrt price is zero
+// - rans out of ticks during swap (token in is too high for liquidity in the pool)
+func (r *routableConcentratedPoolImpl) CalculateTokenInByTokenOut(ctx context.Context, tokenOut sdk.Coin) (sdk.Coin, error) {
```

Comment:
Why do we need to implement this too? I thought we concluded that we can bypass implementing these

Reply:
This one was actually unused just to demonstrate how much extra code it would cost to implement rest of the relevant methods specific to the swap method. I have removed it.